src/Interpreters/InterpreterModifyEngineQuery.cpp

```diff
@@ -0,0 +1,222 @@
+#include <Interpreters/InterpreterModifyEngineQuery.h>
+
+#include <Databases/IDatabase.h>
+#include <Databases/DatabaseOnDisk.h>
+#include <Parsers/ASTModifyEngineQuery.h>
+#include <Interpreters/AddDefaultDatabaseVisitor.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/FunctionNameNormalizer.h>
+#include <Interpreters/QueryLog.h>
+#include <Interpreters/executeQuery.h>
+#include <Interpreters/executeDDLQueryOnCluster.h>
+#include <Interpreters/InterpreterFactory.h>
+#include <Storages/StorageMergeTree.h>
+#include <Storages/StorageReplicatedMergeTree.h>
+#include <Storages/IStorage.h>
+#include <Storages/StorageFactory.h>
+#include <Common/Exception.h>
+#include <Common/typeid_cast.h>
+#include <Access/Common/AccessType.h>
+
+#include <boost/algorithm/string/replace.hpp>
+#include <fmt/core.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+    extern const int BAD_ARGUMENTS;
+    extern const int UNKNOWN_TABLE;
+}
+
+constexpr const char * const REPLICATED_PREFIX = "Replicated";
+
+InterpreterModifyEngineQuery::InterpreterModifyEngineQuery(const ASTPtr & query_ptr_, ContextMutablePtr context_) : WithMutableContext(context_), query_ptr(query_ptr_)
+{
+}
+
+static void setReplicatedEngine(ASTCreateQuery * create_query, ContextPtr context)
+{
+    auto * storage = create_query->storage;
+
+    /// Get replicated engine
+    const auto & config = context->getConfigRef();
+    String replica_path = StorageReplicatedMergeTree::getDefaultZooKeeperPath(config);
+    String replica_name = StorageReplicatedMergeTree::getDefaultReplicaName(config);
+
+    auto args = std::make_shared<ASTExpressionList>();
+    args->children.push_back(std::make_shared<ASTLiteral>(replica_path));
+    args->children.push_back(std::make_shared<ASTLiteral>(replica_name));
+
+    /// Add old engine's arguments
+    if (storage->engine->arguments)
+    {
+        for (size_t i = 0; i < storage->engine->arguments->children.size(); ++i)
+            args->children.push_back(storage->engine->arguments->children[i]->clone());
+    }
+
+    auto engine = std::make_shared<ASTFunction>();
+    engine->name = REPLICATED_PREFIX + storage->engine->name;
+    engine->arguments = args;
+
+    /// Set new engine for the old query
+    create_query->storage->set(create_query->storage->engine, engine);
+}
+
+static void setNotReplicatedEngine(ASTCreateQuery * create_query, ContextPtr)
```

Comment:
What is the usecase? Also, when converting from `ReplicatedMergeTree` to `MergeTree`, it's usually not enough to simply update the metadata file, we should remove table nodes from ZooKeeper as well. And there are some tricky cases, for example, if the `ReplicatedMergeTree` was using zero-copy-replication